Adyen/Analytics/Models/SyncAnalyticsEventDataSource.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+internal protocol AnyAnalyticsEventDataSource {
+    
+    func add(info: AnalyticsEventInfo)
+    func add(log: AnalyticsEventLog)
+    func add(error: AnalyticsEventError)
+    
+    func removeAllEvents()
+    
+    func wrappedEvents() -> AnalyticsEventWrapper?
+}
+
+internal struct AnalyticsEventWrapper {
+    internal var infos: [AnalyticsEventInfo]
+    internal var logs: [AnalyticsEventLog]
+    internal var errors: [AnalyticsEventError]
+}
+
+/// Composes handling of events in a thread safe manner.
+internal final class SyncAnalyticsEventDataSource: AnyAnalyticsEventDataSource {
```

Comment:
Oh now I see the short `sync` version reads as a verb, this idea from myself wasn't the greatest :) 
Maybe `Synchronous`?

Reply:
Other than that this class is seriously beautiful! 🤩

Reply:
`ThreadSafeAnalyticsEventDataSource` (Naming is hard :D)

Reply:
Funny, this was the very first name we came up with! 🤣